azurestack/data_source_subnet.go

```diff
@@ -0,0 +1,102 @@
+package azurestack
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurestack/azurestack/utils"
+)
+
+func dataSourceArmSubnet() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmSubnetRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"virtual_network_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
```

Comment:
This should use the shared schema: `"resource_group_name": resourceGroupNameForDataSourceSchema(),`